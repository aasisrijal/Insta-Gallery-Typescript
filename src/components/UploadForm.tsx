import React, { useState } from "react";

import ProgressBar from "./ProgressBar";

const UploadForm:React.FC = () => {
  const [file, setFile] = useState<File | undefined>();
  const [error, setError] = useState<string | null>(null);

  const types = ["image/png", "image/jpeg"];

  const setFunction = (file: File | undefined) => {
      setFile(file)
  }

  const changeHandler = (e:React.ChangeEvent<HTMLInputElement>) => {
    if (e.target.files === null) {
        return
      }
    let selected = e.target.files[0];
    if (selected && types.includes(selected.type)) {
      setFile(selected);
      setError("");
    } else {
      setFile(undefined);
      setError("Please select an image file of png or jpeg");
    }
  };
  return (
    <form>
        {error && <h3>{error}</h3>}
      <label>
        <input type="file" onChange={changeHandler} />
        <span>+</span>
      </label>
      <div className="output">
        {error && <div className="error">{error}</div>}
        {file && <div>{file.name}</div>}
        {file && <ProgressBar file={file} setFile={setFunction} />}
      </div>
    </form>
  );
};

export default UploadForm;
