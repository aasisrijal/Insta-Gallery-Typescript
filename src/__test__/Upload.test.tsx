import TestRenderer from 'react-test-renderer';
import UploadForm from '../components/UploadForm';

it('renders correctly or not for upload', () => {
    const tree = TestRenderer.create(<UploadForm />).toJSON();
    expect(tree).toMatchSnapshot();
});
