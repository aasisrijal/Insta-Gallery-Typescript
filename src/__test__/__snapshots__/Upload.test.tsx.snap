// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly or not 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="file"
    />
    <span>
      +
    </span>
  </label>
  <div
    className="output"
  />
</form>
`;

exports[`renders correctly or not 2`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="file"
    />
    <span>
      +
    </span>
  </label>
  <div
    className="output"
  />
</form>
`;

exports[`renders correctly or not foir 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="file"
    />
    <span>
      +
    </span>
  </label>
  <div
    className="output"
  />
</form>
`;

exports[`renders correctly or not for modal 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="file"
    />
    <span>
      +
    </span>
  </label>
  <div
    className="output"
  />
</form>
`;

exports[`renders correctly or not for upload 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="file"
    />
    <span>
      +
    </span>
  </label>
  <div
    className="output"
  />
</form>
`;

exports[`renders correctly when there are no items 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      type="file"
    />
    <span>
      +
    </span>
  </label>
  <div
    className="output"
  />
</form>
`;
